pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -434,6 +434,40 @@ func (ds *Plugin) PruneJoinTokens(ctx context.Context, expiry time.Time) (err er
 	})
 }
 
+// CreateFederationRelationship creates a new federation relationship. If the bundle endpoint
+// profile is 'https_spiffe' and the given federation relationship contains a bundle, the current
+// stored bundle is overriden.
+// If no bundle is provided and there is not a previusly stored bundle in the datastore, the
+// federation relationship is not created.
```

Comment:
Note for the reviewer: This is due to [this requirement](https://github.com/spiffe/spire-api-sdk/blob/70fb6a116a63276b5b255ef1485fb261c672078b/proto/spire/api/types/federationrelationship.proto#L29). The bundle update is done in the datastore layer to ensure the operation runs within the same transaction.